--- a/src/panfrost/lib/pan_props.c
+++ b/src/panfrost/lib/pan_props.c
@@ -65,6 +65,7 @@
         MODEL(0x7211, 0, "G76",    "TNOx", HAS_ANISO,         16384, {}),
         MODEL(0x7212, 0, "G52",    "TGOx", HAS_ANISO,         16384, {}),
         MODEL(0x7402, 0, "G52 r1", "TGOx", HAS_ANISO,         16384, {}),
+        MODEL(0x9080, 0, "G77",    "TTRx", HAS_ANISO,         32768, {}),
         MODEL(0x9091, 0, "G57",    "TNAx", HAS_ANISO,         16384, {}),
         MODEL(0x9093, 0, "G57",    "TNAx", HAS_ANISO,         16384, {}),
 
